conans/client/tools/scm.py

```diff
@@ -61,15 +61,58 @@ def run(self, command):
                     else:
                         return self._runner(command)
 
+    def handle_scp_pattern(self, user, domain, url):
```

Comment:
If not local to the function `get_url_with_credentials`, at least make them _private_

```suggestion
    def _handle_scp_pattern(self, user, domain, url):
```